controllers/workspace/provision/routing.go

```diff
@@ -144,15 +145,14 @@ func getSpecRouting(
 
 	var annotations map[string]string
 	if val, ok := workspace.Annotations[config.WorkspaceRestrictedAccessAnnotation]; ok {
-		annotations = map[string]string{}
-		annotations[config.WorkspaceRestrictedAccessAnnotation] = val
+		maputils.Append(annotations, config.WorkspaceRestrictedAccessAnnotation, val)
 	}
 
 	// copy the annotations for the specific routingClass from the workspace object to the routing
 	expectedAnnotationPrefix := workspace.Spec.RoutingClass + config.RoutingAnnotationInfix
 	for k, v := range workspace.GetAnnotations() {
 		if strings.HasPrefix(k, expectedAnnotationPrefix) {
-			annotations[k] = v
+			maputils.Append(annotations, k, v)
```

Comment:
Technically it should work but as long as map is initialized before. So, safer option is to do the following, which is forced by common Append for slices.
```suggestion
			annotations = maputils.Append(annotations, k, v)
```

Reply:
Squashed into b40bf35eb061c12ae61b1825b072ef2778a44886